### PR TITLE
Only admins can change data.

### PR DESCRIPTION
**Pull request by johnnyoshika**
@jsedlacek: only users in role `Admin` are allowed to change data now...all other logged in users have read-only permission.

To support this, I created a `session` service. `session.user` gets set on `afterModel` of `app/admin/route.js` and `session` is injected everywhere where we need to hide UI elements based on role.

Thoughts on this approach?

### CODE REVIEW

File: app/admin/route.js [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
+  session: Ember.inject.service(),
+
   model() {
     return this.store.ajax({
       url: '/api/v1.0/users/me'
     });
   },
-
+  
+  afterModel() {
```
Comment (jsedlacek) [inline app/admin/route.js]: afterModel accepts model as parameter. No need to call `this.modelFor(this.routeName))`.

**Comment by jsedlacek** [comment]
Overall looks good.

I would consider injecting the session service automatically into all controllers using `application.inject`. We probably want to use it almost in any admin `controller`.

File: app/admin/route.js (hunk 1 above)
Comment (johnnyoshika) [inline app/admin/route.js]: Good point.  Code has been changed to use injected model.

**Comment by johnnyoshika** [comment]
@jsedlacek I'm now [injecting session into routes, controllers, and components](https://github.com/bcjobs/analytics-dashbo]rd/commit/46fa1f68079cc46aa8709631d79bf8d7feb7d136).  I noticed that there is a new concept called `instance initializers`, but I used the old `initializer`.  Since we're not using FastBoot, I don't think it matters which one I use, but is one better than the other in this instance?

**Comment by jsedlacek** [comment]
@johnnyoshika Seems fine to me. I have never used the new concept. The documentation is not completely clear in which cases the new form should be used, but until you have multiple application instances it should not matter.

**Pull request merged**